### PR TITLE
Added .editorconfig

### PR DESCRIPTION
**Pull request by xperiandri**

### CODE REVIEW

**Comment by xperiandri** [comment]
@ivelten, @johnberzy-bazinga do you agree with these changes?

**Comment by ivelten** [comment]
@xperiandri I like to use this exact style in my .editorconfig files for .NET solutions. I think we could have this file to help maintain code formatting standards.

File: .editorconfig [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+insert_final_newline     = true
+trim_trailing_whitespace = true
+# Project files
+[*.{csproj,fsproj}]
+indent_size = 2
+charset     = utf-8-bom
+# Code files
+[*.{cs,csx,fs,fsx,vb,vbx,xml}]
+indent_size = 4
+charset     = utf-8-bom
```
Comment (aggieben) [inline .editorconfig]: Why BOM instead of just utf-8?

Comment (xperiandri) [reply .editorconfig]: Usually, I've seen this option set to BOM

**Comment by xperiandri** [comment]
Close in favor of #323

**Comment by xperiandri** [comment]
@ivelten, @mickhansen, @nikhedonia

1. What do you think about the proposed settings?
2. Do you prefer the first or the second commit settings?
3. Would you like anything to add? Maybe about JavaScript

**Comment by xperiandri** [comment]
Now the right commits are in the right place

**Comment by aggieben** [comment]
This seems fine to me 👍 (with the caveat that I can't fully visualize what everything would look like just from the settings).

**Comment by xperiandri** [comment]
Maybe you have your favourite preset 🙂

**Comment by xperiandri** [comment]
To be honest I have not checked for recent new settings in Fantomas. So maybe I miss some of them

**Comment by mickhansen** [comment]
@xperiandri I have no strong feelings about this :)

**Comment by njlr** [comment]
Perhaps we should copy FSharp itself? https://github.com/dotnet/fsharp/blob/main/.editorconfig

**Comment by xperiandri** [comment]
@njlr good idea. I had a look into into it and borrowed 1 settings

**Comment by xperiandri** [comment]
So the latest commit has settings aligned with the latest Fantomas config ([see](https://github.com/fsprojects/fantomas/blob/master/docs/Documentation.md#configuration))

**Comment by xperiandri** [comment]
If no objections are provided today tomorrow I will update all the files according to config

File: .editorconfig [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+# Project files
+[*.{csproj,fsproj,vbproj,shproj,projitems}]
+indent_size          = 2
+charset              = utf-8-bom
+
+###############################
+# F# Coding Conventions       #
+###############################
+[*.{fs,fsx}]
+max_line_length = 120
```
Comment (nojaf) [inline .editorconfig]: Please don't add all the possible Fantomas settings.
This is not recommended, https://github.com/fsprojects/fantomas/blob/master/docs/Documentation.md#configuration

> Please note that you should only add settings to the .editorconfig file when you want to deviate from the default settings. Copying the entire list above is unnecessary.

Comment (xperiandri) [reply .editorconfig]: Thank you for the suggestion!

**Pull request merged**